### PR TITLE
Fix race localizing CategoryAttribute category

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/21356

This is the only explanation I can think of for this failure.

### CODE REVIEW

File: src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs [hunk 1]
```diff
@@ -28,6 +28,8 @@ public class CategoryAttribute : Attribute
 
         private bool _localized;
 
+        private Object _locker = new Object();
```
Comment (safern) [inline src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs]: Nit: Object => object

File: src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs [hunk 2]
```diff
@@ -261,7 +263,6 @@ public CategoryAttribute() : this(nameof(Default))
         public CategoryAttribute(string category)
         {
             _categoryValue = category;
-            _localized = false;
```
Comment (safern) [inline src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs]: Why is this removed?

Comment (danmosemsft) [inline src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs]: I'ts the default value, so the runtime initializes it for you anyway.

File: src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs]: Oops, fixed. My last team had the opposite style.

File: src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs (hunk 2 above)
Comment (safern) [inline src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs]: Oh you are right, missed it was in the constructor :)

**Comment by hughbe** [comment]
Here I was thinking it was a test bug. Thanks
